core/state_processor.go

```diff
@@ -86,8 +86,12 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 		receipts = append(receipts, receipt)
 		allLogs = append(allLogs, receipt.Logs...)
 	}
+	// Fail if Shanghai not enabled and len(withdrawals) is non-zero.
+	if !p.config.IsShanghai(new(big.Int).SetUint64(block.Time())) && block.Withdrawals() != nil {
```

Comment:
Felix thinks we should just check the length here, not the nil-ness. 
Right now, the comment above does not match the behaviour on that point.